exist-core/src/main/java/org/exist/backup/restore/RestoreHandler.java

```diff
@@ -304,11 +304,16 @@ private DeferredPermission restoreResourceEntry(final Attributes atts) throws SA
         }
 
         MimeType mimeType = null;
-        if (mimeTypeStr != null) {
+        if (xmlType) {
+            mimeType = MimeType.XML_TYPE;
+        } else if (mimeTypeStr != null) {
             mimeType = MimeTable.getInstance().getContentType(mimeTypeStr);
-        }
-        if (mimeType == null) {
-            mimeType = xmlType ? MimeType.XML_TYPE : MimeType.BINARY_TYPE;
+            if (mimeType == null) {
+                mimeType = MimeType.BINARY_TYPE;
```

Comment:
this value should probably be taken from the default in mimetypes.xml ?

Reply:
No, as we want to have the resource type as defined within the backup meta data to take precedence over those within the mime type configuration.